src/hidimstat/conditional_feature_importance.py

```diff
@@ -1,80 +1,92 @@
 
 
 class CFI(BasePerturbation):
+    """
+    Conditional Feature Importance (CFI) algorithm.
+    :footcite:t:`Chamma_NeurIPS2023` and for group-level see
+    :footcite:t:`Chamma_AAAI2024`.
+
+    Parameters
+    ----------
+    estimator : sklearn compatible estimator
+        The estimator to use for the prediction.
+    method : str, default="predict"
+        The method to use for the prediction. This determines the predictions passed
+        to the loss function. Supported methods are "predict", "predict_proba" or
+        "decision_function".
+    loss : callable, default=root_mean_squared_error
+        The loss function to use when comparing the perturbed model to the full
+        model.
+    n_permutations : int, default=50
+        The number of permutations to perform. For each variable/group of variables,
+        the mean of the losses over the `n_permutations` is computed.
+    imputation_model_continuous : sklearn compatible estimator, default=RidgeCV()
+        The model used to estimate the conditional distribution of a given
+        continuous variable/group of variables given the others.
+    imputation_model_categorical : sklearn compatible estimator, default=LogisticRegressionCV()
+        The model used to estimate the conditional distribution of a given
+        categorical variable/group of variables given the others. Binary is
+        considered as a special case of categorical.
+    features_groups: dict or None, default=None
+        A dictionary where the keys are the group names and the values are the
+        list of column names corresponding to each features group. If None,
+        the features_groups are identified based on the columns of X.
+    feature_types: str or list, default="auto"
+        The feature type. Supported types include "auto", "continuous", and
+        "categorical". If "auto", the type is inferred from the cardinality
+        of the unique values passed to the `fit` method.
+    categorical_max_cardinality : int, default=10
+        The maximum cardinality of a variable to be considered as categorical
+        when the variable type is inferred (set to "auto" or not provided).
+    test_statistic : callable, default=partial(wilcoxon, axis=1)
+        Statistical test function used to compute p-values for importance scores.
+        Must accept an array of values and return an object with a 'pvalue' attribute.
+        Default is Wilcoxon signed-rank test.
+    random_state : int or None, default=None
+        The random state to use for sampling.
+    n_jobs : int, default=1
+        The number of jobs to run in parallel. Parallelization is done over the
+        variables or groups of variables.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+
     def __init__(
         self,
         estimator,
-        loss: callable = root_mean_squared_error,
         method: str = "predict",
+        loss: callable = root_mean_squared_error,
```

Comment:
As discussed with @AngelReyero can we take this opportunity to replace `root_mean_squared_error` by `mean_squared_error`

Reply:
done